sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala

```diff
@@ -89,6 +97,35 @@ case class AnalyzeColumnCommand(
     columnsToAnalyze
   }
 
+  private def analyzeColumnInCatalog(sparkSession: SparkSession): Unit = {
+    val sessionState = sparkSession.sessionState
+    val tableMeta = sessionState.catalog.getTableMetadata(tableIdent)
+    if (tableMeta.tableType == CatalogTableType.VIEW) {
+      throw new AnalysisException("ANALYZE TABLE is not supported on views.")
```

Comment:
Since this PR can analyze all cached plans, we need to remove this exception logically.
For example, how do you think about the following case? Currently, this fails due to this. It would be great if we can handle this too.
```scala
  test("database view") {
    withDatabase("db_v") {
      sql("CREATE DATABASE db_v")
      sql("CREATE VIEW db_v.v1 AS SELECT 1 c1")
      sql("CACHE TABLE db_v.v1")
      sql("ANALYZE TABLE db_v.v1 COMPUTE STATISTICS FOR COLUMNS c1")
      sql("SELECT * FROM db_v.v1").show
    }
  }
```

Reply:
It seems we can do the same fix for AnalyzeTableCommand: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala#L38
If no problem, I'll do in a separate follow-up pr.